packages/builder/src/components/backend/Datasources/CreateEditRelationship.svelte

```diff
@@ -59,6 +76,10 @@
     if ($touched.through && isMany && !fromRelate.through) {
       errObj.through = tableNotSet
     }
+    if ($touched.through && invalidThroughTable(fromRelate)) {
+      errObj.through =
+        "Table contains non-nullable columns which aren't generated"
```

Comment:
I would potentially flip this to make it easier to understand:

`Table PK must have auto generated` or similar.

Reply:
In this case its specifically about non-nullable fields, its not to do with the PK. For example you could have a whole bunch of non-nullable fields in your junction table, Budibase will never be able to fill them in - creation of rows will always fail.

Reply:
Gotcha, in that case maybe:
`Ensure all columns in table are nullable or auto generated`